rotkehlchen/tests/api/test_history_export.py

```diff
@@ -0,0 +1,152 @@
+        api_url_for(rotkehlchen_api_server_with_exchanges, 'exporthistoryeventresource'),
+        json={
+            'directory_path': csv_dir,
+        },
+    )
+    assert_csv_export_response(response, csv_dir)
+
+    # now query the export endpoint with query params
+    response = requests.post(api_url_for(
+        rotkehlchen_api_server_with_exchanges,
+        'exporthistoryeventresource',
+        directory_path=csv_dir2,
+        from_timestamp=1500000000,
+        to_timestamp=1600000000,
+    ))
+    assert_csv_export_response(response, csv_dir2, 2)
+
+    # now query the download CSV endpoint
+    response = requests.patch(
+        api_url_for(rotkehlchen_api_server_with_exchanges, 'exporthistoryeventresource'),
+    )
+    temp_csv_file = Path(download_dir, FILENAME_HISTORY_EVENTS_CSV)
+    temp_csv_file.write_bytes(response.content)
+    assert_csv_export_response(response, download_dir, is_download=True)
+
+
+def test_history_export_csv_errors(
+        rotkehlchen_api_server_with_exchanges,
+        tmpdir_factory,
+):
+    """Test that errors on the csv export REST API endpoint are handled correctly"""
+    rotki = rotkehlchen_api_server_with_exchanges.rest_api.rotkehlchen
+    prepare_rotki_for_history_processing_test(
+        rotki,
+        should_mock_history_processing=False,
+    )
+    csv_dir = str(tmpdir_factory.mktemp('test_csv_dir'))
+
+    # Query the export endpoint without first having queried the history
+    response = requests.post(
+        api_url_for(rotkehlchen_api_server_with_exchanges, 'exporthistoryeventresource'),
+        json={'directory_path': csv_dir},
+    )
+    assert_error_response(
+        response=response,
+        contained_in_msg='No history processed in order to perform an export',
+        status_code=HTTPStatus.CONFLICT,
+    )
+
+    # Now, add data for exporting
+    rotki = rotkehlchen_api_server_with_exchanges.rest_api.rotkehlchen
+    db = DBHistoryEvents(rotki.data.db)
+    add_entries(server=rotkehlchen_api_server_with_exchanges, events_db=db, add_directly=True)
+
+    # And now provide non-existing path for directory
+    response = requests.post(
+        api_url_for(rotkehlchen_api_server_with_exchanges, 'exporthistoryeventresource'),
+        json={'directory_path': '/idont/exist/for/sure/'},
+    )
+    assert_error_response(
+        response=response,
+        contained_in_msg='"directory_path": ["Given path /idont/exist/for/sure/ does not exist',
+        status_code=HTTPStatus.BAD_REQUEST,
+    )
+
+    # And now provide valid path but not directory
+    temporary_file = Path(Path(csv_dir) / 'f.txt')
+    temporary_file.touch()
+    response = requests.post(
+        api_url_for(rotkehlchen_api_server_with_exchanges, 'exporthistoryeventresource'),
+        json={'directory_path': str(temporary_file)},
+    )
+    assert_error_response(
+        response=response,
+        contained_in_msg='is not a directory',
+        status_code=HTTPStatus.BAD_REQUEST,
+    )
```

Comment:
I need you to add a new test in this file, or augment the first one.

That test should export all possible types of history events to see that exports are done fine for all types of history events. That means, the basic one, evm events, eth deposits, eth withdrawals, eth block production etc.

It's mainly to test that the headers are done correctly and that all the attributes, of all events are exported succesfully. Also keep in mind we will add a lot more history event types so this should work flawlessly